sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -151,7 +151,7 @@ case class CreateDataSourceTableAsSelectCommand(
         table.storage.locationUri
       }
       val result = saveDataIntoTable(
-        sparkSession, table, tableLocation, query, mode, tableExists = false)
+        sparkSession, table, tableLocation, query, mode, overwrite = true, tableExists = false)
```

Comment:
in this branch, table not exist, and the expected behavior is to create a directory or overwrite it if it's already existed. So we can just pass a `OverWrite` mode here, and everything should work. Did I miss something?